### PR TITLE
EntryLinkHelpers.ContentLink should be available to use for user code

### PR DESCRIPTION
**Pull request by pranavkm**
inside of a cache tag helper's body.

Fixes: #1867

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Tratcher. Could you have a look and make sure I'm doing the right thing?

File: samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            _moviesService = moviesService;
+        }
+
+        public IViewComponentResult Invoke()
+        {
+            IExpirationTrigger trigger;
+            var movies = _moviesService.GetFeaturedMovies(out trigger);
+            
+            // Add custom triggers
+            EntryLinkHelpers.ContextLink.AddExpirationTriggers(new[] { trigger });
```
Comment (davidfowl) [inline samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs]: That's an ugly API

Comment (pranavkm) [inline samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs]: Yup.

Comment (davidfowl) [inline samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs]: Why is it so ugly?

**Comment by Tratcher** [comment]
Looks ok.

File: samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs (hunk 1 above)
Comment (pranavkm) [inline samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs]: Because I didn't write it :dancer:

Comment (Tratcher) [inline samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs]: We didn't expect developers to access it directly.

**Comment by NTaylorMullen** [comment]
TagHelper side of things look good :+1:

File: samples/TagHelperSample.Web/Services/MoviesService.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            _quotesTokenSource.Dispose();
+            _quotesTokenSource = null;
+        }
+
+        private IEnumerable<FeaturedMovies> GetMovies()
+        {
+            yield return new FeaturedMovies { Name = "A day in the life of a blue whale", Rank = _random.Next(1, 10) };
+            yield return new FeaturedMovies { Name = "FlashForward", Rank = _random.Next(1, 10) };
+            yield return new FeaturedMovies { Name = "Frontier", Rank = _random.Next(1, 10) };
+            yield return new FeaturedMovies { Name = "Attack of the space spiders", Rank = _random.Next(1, 10) };
```
Comment (rynowak) [inline samples/TagHelperSample.Web/Services/MoviesService.cs]: `_random.Next(9.9, 10)`

File: test/WebSites/MvcTagHelpersWebSite/ProductsService.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public class ProductsService
+    {
+        private readonly CancellationTokenSource _tokenSource = new CancellationTokenSource();
+
+        public string GetProducts(string category, out IExpirationTrigger trigger)
+        {
+            trigger = new CancellationTokenTrigger(_tokenSource.IsCancellationRequested ? CancellationToken.None :
+                                                                                          _tokenSource.Token);
+
+            if (category == "Books")
```
Comment (rynowak) [inline test/WebSites/MvcTagHelpersWebSite/ProductsService.cs]: extra blank line

File: test/WebSites/MvcTagHelpersWebSite/ProductsService.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace MvcTagHelpersWebSite
+{
+    public class ProductsService
+    {
+        private readonly CancellationTokenSource _tokenSource = new CancellationTokenSource();
+
+        public string GetProducts(string category, out IExpirationTrigger trigger)
+        {
+            trigger = new CancellationTokenTrigger(_tokenSource.IsCancellationRequested ? CancellationToken.None :
```
Comment (rynowak) [inline test/WebSites/MvcTagHelpersWebSite/ProductsService.cs]: hard to read, beak this up

Comment (rynowak) [inline test/WebSites/MvcTagHelpersWebSite/ProductsService.cs]:
```
var shouldIUseATernary = isReadable ? "Maybe" : "Definitely Not.";
```

**Comment by rynowak** [comment]
:shipit:  after fixing minor comments.

I share the same concerns about the usability/discoverability of this. Product code looks fine.

**Pull request closed without merging**